### PR TITLE
[Custom log types] Show log types table, Log type creation workflow

### PR DESCRIPTION
**Pull request by amsiglan**
### Description
In this PR we have added below workflows in UX:
- Create custom log types
- Update log types
- List log types in a table

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/store/LogTypeStore.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    const logTypesRes = await this.service.searchLogTypes(id);
+    if (logTypesRes.ok) {
+      const logTypes: LogType[] = logTypesRes.response.hits.hits.map((hit) => {
+        return {
+          id: hit._id,
+          ...hit._source,
+        };
+      });
+
+      return { ...logTypes[0], detectionRules: 10 };
```
Comment (lezzago) [inline public/store/LogTypeStore.ts:24]: Why are we hard coding `detectionRules: 10` to 10?

Comment (lezzago) [reply public/store/LogTypeStore.ts:24]: Also if it needs to be hard coded, should this be a constant?

File: server/services/LogTypeService.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<
+    IOpenSearchDashboardsResponse<ServerResponse<UpdateLogTypeResponse> | ResponseError>
+  > => {
+    try {
+      const logType = request.body;
+      const { logTypeId } = request.params;
+      const params: UpdateLogTypeParams = { body: logType, logTypeId };
+      const { callAsCurrentUser: callWithRequest } = this.osDriver.asScoped(request);
+      console.log(JSON.stringify(params));
```
Comment (lezzago) [inline server/services/LogTypeService.ts:117]: dev code?

**Comment by lezzago** [comment]
Overall LGTM, just need the unit tests to pass. Also not sure if cypress tests need to be updated or they have previous issues.

File: public/store/LogTypeStore.ts (hunk 1 above)
Comment (amsiglan) [reply public/store/LogTypeStore.ts]: Changed logic to fetch the count

File: server/services/LogTypeService.ts (hunk 2 above)
Comment (amsiglan) [reply server/services/LogTypeService.ts]: Removed

**Comment by amsiglan** [comment]
> Overall LGTM, just need the unit tests to pass. Also not sure if cypress tests need to be updated or they have previous issues.

Updated snapshots, UTs should pass now. Looking into cypress tests

File: .github/workflows/cypress-workflow.yml [hunk 3]
```diff
@@ -7,14 +7,14 @@ on:
     branches:
       - "*"
 env:
-  OPENSEARCH_DASHBOARDS_VERSION: '2.8.0'
-  OPENSEARCH_VERSION: '2.8.0-SNAPSHOT'
+  OPENSEARCH_DASHBOARDS_VERSION: '2.9.0'
+  OPENSEARCH_VERSION: '2.9.0-SNAPSHOT'
```
Comment (lezzago) [inline .github/workflows/cypress-workflow.yml:11]: should this be 3.0 since its main?

**Review by lezzago — APPROVED** [review]

File: .github/workflows/cypress-workflow.yml (hunk 3 above)
Comment (amsiglan) [reply .github/workflows/cypress-workflow.yml:11]: There were issues with main backend due to which cypress tests fail to run, will create an issue to get that fixed and revert this to main

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**